### ISSUE TITLE
Multiclass Classification Support is Broken

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1848]
Multiclass support in DeepChem is currently broken. There are a number of places where classification models assume binary data and make shape casts accordingly. Here's an (incomplete) list of places with this issue:

- `dc.trans.BalancingTransformer` isn't multiclass aware (#1813)
- `Model.predict()` doesn't work correctly for multiclass data (it seems to assume the data is binary and just returns the 0-class probability).
- probably others...

H/T to @ramanktalwar who dug out many of these failures! We should try to fix these before the 2.4 release

**Comment by rbharath** [comment]
As a couple other notes, `GraphConvModel.fit()` (and probably the general `KerasModel.fit()`) don't work for multiclass data.

**Comment by rbharath** [comment]
As a second note, `Chemception` models also fail with multiclass prediction/evaluation.

**Comment by vsomnath** [comment]
https://github.com/deepchem/deepchem/blob/51f426b3b6acb876703e49d24ef4f16cb0b44b43/deepchem/models/chemnet_models.py#L278

The ChemCeption model switches between Softmax and Sigmoid, depending on number of classes.  Strange they run into multiclass issues.

**Comment by nd-02110114** [comment]
@rbharath 
I seem `GraphConvModel` work well for multiclass data. What is the problem...?

**Comment by rbharath** [comment]
Ah this was fixed in my refactoring of the metrics a few weeks ago :). Let me go ahead and close this issue!
